web/crux/src/app/registry/registry.dto.ts

```diff
@@ -0,0 +1,151 @@
+import { Type } from 'class-transformer'
+import { IsDate } from 'class-validator'
+
+export const REGISTRY_TYPE_VALUES = ['v2', 'hub', 'gitlab', 'github', 'google', 'unchecked'] as const
+export type RegistryType = (typeof REGISTRY_TYPE_VALUES)[number]
+
+export const GITHUB_NAMESPACE_VALUES = ['organization', 'user'] as const
+export const GITLAB_NAMESPACE_VALUES = ['group', 'project'] as const
+export type GitlabNamespace = (typeof GITLAB_NAMESPACE_VALUES)[number]
+export type GithubNamespace = (typeof GITHUB_NAMESPACE_VALUES)[number]
+export type RegistryNamespace = GitlabNamespace | GithubNamespace
+
+export class RegistryResponse {
+  id: string
+
+  name: string
+
+  description?: string
+
+  icon?: string
+
+  url: string
+
+  type: RegistryType
+}
+
+export class RegistryListResponse {
+  data: RegistryResponse[]
+}
+
+export class HubRegistryDetails {
+  imageNamePrefix: string
+}
+
+export class V2RegistryDetails {
+  url: string
+
+  user?: string
+
+  token?: string
+}
+
+export class GitlabRegistryDetails {
+  user: string
+
+  token: string
+
+  imageNamePrefix: string
+
+  url?: string
+
+  apiUrl?: string
+
+  namespace: RegistryNamespace
+}
+
+export class GithubRegistryDetails {
+  user: string
+
+  token: string
+
+  imageNamePrefix: string
+
+  namespace: RegistryNamespace
+}
+
+export class GoogleRegistryDetails {
+  url: string
+
+  user?: string
+
+  token?: string
+
+  imageNamePrefix: string
+}
+
+export class UncheckedRegistryDetails {
+  url: string
+}
+
+export class RegistryDetailsResponse {
+  id: string
+
+  name: string
+
+  description?: string
+
+  icon?: string
+
+  inUse: boolean
+
+  @Type(() => Date)
+  @IsDate()
+  createdAt: Date
+
+  @Type(() => Date)
+  @IsDate()
+  updatedAt: Date
+
+  hub?: HubRegistryDetails
+
+  v2?: V2RegistryDetails
+
+  gitlab?: GitlabRegistryDetails
+
+  github?: GithubRegistryDetails
+
+  google?: GoogleRegistryDetails
+
+  unchecked?: UncheckedRegistryDetails
+}
+
+export class CreateRegistryRequest {
+  name: string
+
+  description?: string
+
+  icon?: string
+
+  hub?: HubRegistryDetails
+
+  v2?: V2RegistryDetails
+
+  gitlab?: GitlabRegistryDetails
```

Comment:
These works with Swagger without `@ApiProperty`? Did you check Swagger?

Reply:
Yes they show up with all of their fields.

Reply:
Good to hear, comment resolved.